packages/menu-surface/index.tsx

```diff
@@ -253,62 +250,60 @@ class MenuSurface extends React.Component<MenuSurfaceProps, MenuSurfaceState> {
       getWindowScroll: () => {
         return {x: window.pageXOffset, y: window.pageYOffset};
       },
-      setPosition: (position: Position) => {
+      setPosition: (position: Partial<MDCMenuDistance>) => {
         this.setState((prevState) => Object.assign(prevState, {
           styleLeft: 'left' in position ? position.left : null,
           styleRight: 'right' in position ? position.right : null,
           styleTop: 'top' in position ? position.top : null,
           styleBottom: 'bottom' in position ? position.bottom : null,
         }));
       },
-      setMaxHeight: (maxHeight: number) => this.setState({maxHeight}),
+      setMaxHeight: (maxHeight: string) => this.setState({maxHeight}),
     };
 
-    return Object.assign(
-      {
-        addClass: (className: string) => {
-          const classList = new Set(this.state.classList);
-          classList.add(className);
-          this.setState({classList});
-        },
-        removeClass: (className: string) => {
-          const classList = new Set(this.state.classList);
-          classList.delete(className);
-          this.setState({classList});
-        },
-        hasClass: (className: string) => this.classes.split(' ').includes(className),
-        hasAnchor: () => !!this.props.anchorElement,
-        notifyOpen: () => {
-          if (this.registerWindowClickListener) {
-            this.registerWindowClickListener();
-          }
-          this.props.onOpen!();
-        },
-        notifyClose: () => {
-          if (this.deregisterWindowClickListener) {
-            this.deregisterWindowClickListener();
-          }
-          this.props.onClose!();
-        },
-        isElementInContainer: (el: HTMLElement) => {
-          if (!this.menuSurfaceElement.current) return false;
-          if (this.menuSurfaceElement.current === el) {
-            return true;
-          }
-          return this.menuSurfaceElement.current.contains(el);
-        },
-        isRtl: () => {
-          if (!this.menuSurfaceElement) return false;
-          if (!this.menuSurfaceElement.current) return false;
-          return window
-            .getComputedStyle(this.menuSurfaceElement.current)
-            .getPropertyValue('direction') === 'rtl';
-        },
-        setTransformOrigin: (transformOrigin: string) => this.setState({transformOrigin}),
+    return {
+      addClass: (className: string) => {
+        const classList = new Set(this.state.classList);
+        classList.add(className);
+        this.setState({classList});
       },
-      focusAdapterMethods,
-      dimensionAdapterMethods
-    );
+      removeClass: (className: string) => {
+        const classList = new Set(this.state.classList);
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className: string) => this.classes.split(' ').includes(className),
```

Comment:
This should use `classList` (Set) state for consistency.

Reply:
So the naming might be causing confusion, but `classList` is from the foundationClasses (what I want to rename all of them to). `this.classes` contains all the classes that are on the menu-surface, including `mdc-menu-surface`,  `mdc-menu-surface--fixed`, and anything the foundation or user add to the component.